app/Http/Middleware/SetLocale.php

```diff
@@ -0,0 +1,23 @@
+<?php
+namespace Ushahidi\App\Http\Middleware;
+
+use Closure;
+
+class SetLocale
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @param  mixed  ...$features
+     * @return mixed
+     */
+    public function handle($request, Closure $next, ...$features)
+    {
+        $locale = $request->header('Accept-language');
```

Comment:
@rowasc  how does this work with complex but still valid values of the accept-language header? i.e. `fr-CH, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5` ?

Reply:
semed to work fine, luckily. I actually had the same concern

Reply:
gross no, I think I was half asleep when I checked this. Only supports `en` / `es` etc right now (because all it does is check if the language folder exists, so if there's no `es-ES blablabla` folder it won't find it)

So we need a small parser to tries to find es-ES, if it doesn't it fallbacks to es, and if it doesn't it fall backs to en.
--